lib/CL/devices/basic/basic.c

```diff
@@ -719,14 +719,21 @@ pocl_basic_fill_rect (void *data,
                 + buffer_slice_pitch * k, fill_pixel, pixel_size);
 }
 
+typedef unsigned long v2ul __attribute__ ((vector_size (16)));
```

Comment:
This might be counter productive. The vector datatypes are an extension (although supported both by gcc and Clang) pls I believe the loops should get easily vectorized automatically too.